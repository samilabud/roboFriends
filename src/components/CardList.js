import React from 'react';
import Card from './Card';


const CardList = ({robots}) => {
    const robotListed = robots.map((value, i)=>{
        return <Card key={i} id={robots[i].id} name={robots[i].name} email={robots[i].email}  />
    })
    //ternario
    return !robots.length ? <h2>Sin resultados</h2>
    : (
            <div>
                {robotListed}
            </div>
        );
}

export default CardList;