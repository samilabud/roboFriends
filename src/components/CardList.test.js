import React from 'react';
import { create } from "react-test-renderer";
import CardList from  './CardList';




it('Testing CardList component',()=>{
  //expect(shallow(<Card />)).toMatchSnapshot();
  const mockRobots = [
    {
      id: 1,
      name: "Jhon Snow",
      email: "jhon.snow@zerotomastery.com"
    },
    {
      id: 2,
      name: "Katlin Star",
      email: "katlin.starw@zerotomastery.com"
    },
  ]
  const app = create(<CardList robots={mockRobots} />);
  expect(app.toJSON()).toMatchSnapshot(); 
})