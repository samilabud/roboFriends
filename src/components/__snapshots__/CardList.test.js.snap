// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardList component 1`] = `
<div>
  <div
    className="bg-washed-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robots"
      src="https://robohash.org/1?200x200"
    />
    <div>
      <h2>
        Jhon Snow
      </h2>
      <p>
        jhon.snow@zerotomastery.com
      </p>
    </div>
  </div>
  <div
    className="bg-washed-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robots"
      src="https://robohash.org/2?200x200"
    />
    <div>
      <h2>
        Katlin Star
      </h2>
      <p>
        katlin.starw@zerotomastery.com
      </p>
    </div>
  </div>
</div>
`;
